bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/link/ItemChannelLinkResource.java

```diff
@@ -191,17 +201,14 @@ public Response unlink(@PathParam("itemName") @Parameter(description = "itemName
             String message = "Link " + linkId + " does not exist!";
             return JSONResponse.createResponse(Status.NOT_FOUND, null, message);
         }
-
         ItemChannelLink result = itemChannelLinkRegistry.remove(linkId);
-        if (result != null) {
-            return Response.ok(null, MediaType.TEXT_PLAIN).build();
-        } else {
+        if (result == null) {
             return JSONResponse.createErrorResponse(Status.METHOD_NOT_ALLOWED, "Channel is read-only.");
```

Comment:
Wouldn't that be the link rather than the channel actually?

Reply:
imo sending status 405 is sufficient. The message is not needed at all.